lib/sshutils/server.go

```diff
@@ -441,6 +441,12 @@ func (s *Server) trackUserConnections(delta int32) int32 {
 	return atomic.AddInt32(&s.userConns, delta)
 }
 
+// ActiveConnections returns the number of connections that are
+// being served.
+func (s *Server) ActiveConnections() int32 {
+	return atomic.LoadInt32(&s.userConns)
```

Comment:
Something I don't yet grok fully: I assume we use the atomic package in this struct for better performance over `sync.Mutex`, yet the docs for the atomic package say:

> These functions require great care to be used correctly. Except for special, low-level applications, [sync](https://pkg.go.dev/sync)hronization is better done with channels or the facilities of the sync package.

Despite that, I see atomic operations generally being recommended over mutexes. Here's [an example answer on Stack Overflow](https://stackoverflow.com/a/47446114/742872).

What is it about this specific `Server` struct that guarantees that atomic operations are the better option over mutexes?